addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -12,25 +13,32 @@ publicWidget.registry.s_website_form.include({
             if (!this.isEditable && !this.$('.s_turnstile').length && session.turnstile_site_key) {
+                // global callback for the turnstile script to call
+                // Rethrow the error, or we only will catch a "Script error" without any info 
+                // because of the script api.js originating from a different domain.
+                function throwTurnstileError(code) {
+                    const error = new Error("Turnstile Error");
+                    error.code = code;
+                    throw error;
+                }
+                const turnstileErrorGlobalScript = document.createElement("script");
+                turnstileErrorGlobalScript.classList.add("s_turnstile");
+                turnstileErrorGlobalScript.textContent = throwTurnstileError.toString();
+
                 const mode = new URLSearchParams(window.location.search).get('cf') == 'show' ? 'always' : 'interaction-only';
-                $(`<div class="s_turnstile cf-turnstile float-end"
-                         data-action="website_form"
-                         data-appearance="${mode}"
-                         data-response-field-name="turnstile_captcha"
-                         data-sitekey="${session.turnstile_site_key}"
-                         data-error-callback="throwTurnstileError"
-                    ></div>
-                    <script class="s_turnstile">
-                        // Rethrow the error, or we only will catch a "Script error" without any info 
-                        // because of the script api.js originating from a different domain.
-                        function throwTurnstileError(code) {
-                            const error = new Error("Turnstile Error");
-                            error.code = code;
-                            throw error;
-                        }
-                    </script>
-                    <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
-                `).insertAfter('.s_website_form_send, .o_website_form_send');
+                const turnstileContainer = renderToElement("website_cf_turnstile.turnstile_container", {
+                    action: "website_form",
+                    appearance: mode,
+                    additionalClasses: "float-end",
+                    errorGlobalCallback: throwTurnstileError.name,
+                    sitekey: session.turnstile_site_key,
+                });
+                const turnstileScript = renderToElement("website_cf_turnstile.turnstile_remote_script");
+
+                const sendButton = document.querySelector(".s_website_form_send, .o_website_form_send");
+                sendButton.parentNode.insertBefore(turnstileContainer, sendButton.nextSibling);
+                sendButton.parentNode.insertBefore(turnstileErrorGlobalScript, sendButton.nextSibling);
+                sendButton.parentNode.insertBefore(turnstileScript, sendButton.nextSibling);
```

Comment:
Aren't all these calls simply this:
```suggestion
                sendButton.after(turnstileContainer, turnstileErrorGlobalScript, turnstileScript);
```

Reply:
:o nice, I thought it was weird for there to not be an insertAfter in the dom api :D

patApplied